### PR TITLE
Fix invalid response and rename ListPlanAccountsForAccount (fixes #1270)

### LINKED ISSUE #1270: Invalid response expected for ListPlanAccountsForAccount
Currently `ListPlanAccountsForAccount` from `MarketplaceService` expects `[]*MarketplacePlanAccount`.

But documentation https://developer.github.com/v3/apps/marketplace/#check-if-a-github-account-is-associated-with-any-marketplace-listing says that `GET /marketplace_listing/accounts/:account_id` returns `*MarketplacePlanAccount`.

This is causing my application to fail.

### PR DESCRIPTION
**Pull request by sethaustinx**
To simplify review and address specific requests from https://github.com/google/go-github/issues/1270#issuecomment-791389308 I've made separate commits.

Also wording doesn't seem to match GitHub docs anymore.

If that's ok @gmlewis, I'd like to rename:
- GetPlanAccountForAccount to GetSubscriptionPlanForAccount https://octokit.github.io/rest.js/v18#apps-get-subscription-plan-for-account
- MarketplacePlanAccount to MarketplaceSubscriptionPlan

What do you think?

### CODE REVIEW

**Comment by sethaustinx** [comment]
@googlebot I signed it!

**Comment by gmlewis** [comment]
> If that's ok @gmlewis, I'd like to rename:
> 
> * GetPlanAccountForAccount to GetSubscriptionPlanForAccount https://octokit.github.io/rest.js/v18#apps-get-subscription-plan-for-account
> * MarketplacePlanAccount to MarketplaceSubscriptionPlan
> 
> What do you think?

Yes, those seem like reasonable changes.  Thank you, @sethaustinx .  I'll take a look at your PR.

**Review by gmlewis — APPROVED** [review]
Thank you, @sethaustinx !
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
💙

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Comment by sethaustinx** [comment]
Awesome :orange_heart:

**Pull request merged**
